### ISSUE TITLE
Must provide exactly one storage type error.

### ISSUE THREAD

**Issue by jcnnghm** [docker/distribution#511]
Running `registry:2.0.0` with a custom config, I get the error `Must provide exactly one storage type. Provided: [filesystem maintenance]`.  Seems like this should be addressed by https://github.com/docker/distribution/blob/70427e3d9f1224a2bdc24ed454223f364e9d3820/configuration/configuration.go#L222.  

More complete logs:
```
configuration error: error parsing /registry/config.yml: Must provide exactly one storage type. Provided: [filesystem maintenance]
usage: registry <config>
  -version=false: show the version and exit
configuration error: error parsing /registry/config.yml: Must provide exactly one storage type. Provided: [filesystem maintenance]
usage: registry <config>
  -version=false: show the version and exit
configuration error: error parsing /registry/config.yml: Must provide exactly one storage type. Provided: [filesystem maintenance]
usage: registry <config>
  -version=false: show the version and exit
configuration error: error parsing /registry/config.yml: Must provide exactly one storage type. Provided: [filesystem maintenance]
usage: registry <config>
  -version=false: show the version and exit
```

This happens even when running with the example config from https://raw.githubusercontent.com/docker/distribution/master/cmd/registry/config.yml.

```yaml
version: 0.1
log:
  level: debug
  fields:
    service: registry
    environment: development
storage:
    cache:
        layerinfo: inmemory
    filesystem:
        rootdirectory: /tmp/registry-dev
    maintenance:
        uploadpurging:
            enabled: false
http:
    addr: :5000
    secret: asecretforlocaldevelopment
    debug:
        addr: localhost:5001
redis:
  addr: localhost:6379
  pool:
    maxidle: 16
    maxactive: 64
    idletimeout: 300s
  dialtimeout: 10ms
  readtimeout: 10ms
  writetimeout: 10ms
notifications:
    endpoints:
        - name: local-8082
          url: http://localhost:5003/callback
          headers:
             Authorization: [Bearer <an example token>]
          timeout: 1s
          threshold: 10
          backoff: 1s
          disabled: true
        - name: local-8083
          url: http://localhost:8083/callback
          timeout: 1s
          threshold: 10
          backoff: 1s
          disabled: true
```

**Comment by RichardScothern** [comment]
hi @jcnnghm
Upload purge configuration is not implemented in 2.0.0 and therefore the configuration will be rejected.  You are right that the code in master which you refer to handles this case.

To work around this problem you can remove the `maintenance` section from your configuration.  Sorry this wasn't clearer,

**Comment by dmp42** [comment]
@stevvooe @RichardScothern dealing with confusion wr t"master documentation/configs being what you get when you browse github vs. people using a released version" is an unsolved issue... I don't have the right answer (had to resort to fat warnings in the past: https://github.com/hipache/hipache#warning).
